Python/codegen.c

```diff
@@ -6019,11 +6030,18 @@ codegen_pattern_value(compiler *c, pattern_ty p, pattern_context *pc)
 {
     assert(p->kind == MatchValue_kind);
     expr_ty value = p->v.MatchValue.value;
-    if (!MATCH_VALUE_EXPR(value)) {
+    // The permitted expressions in a case pattern are constants,
+    // attribute lookups, and complex literals. However,
+    // complex literals are represented as a binary operation in
+    // the AST rather than a constant, so we need to check for them
+    // manually here.
+    if (MATCH_VALUE_EXPR(value) || is_complex_binop(value)) {
+        VISIT(c, expr, value);
+    }
+    else {
         const char *e = "patterns may only match literals and attribute lookups";
         return _PyCompile_Error(c, LOC(p), e);
     }
```

Comment:
```suggestion
    if (!MATCH_VALUE_EXPR(value) && !is_complex_binop(value)) {
        const char *e = "patterns may only match literals and attribute lookups";
        return _PyCompile_Error(c, LOC(p), e);
    }
    VISIT(c, expr, value);
```
Less diff & simpler flow.

Reply:
I think I can combine both into a single inline function

Reply:
I think it makes sense because there will be also unaryop check, right @Eclips4 ?

Reply:
> I think it makes sense because there will be also unaryop check, right @Eclips4 ?

Yup!

Reply:
Done!